python/vtool/ramen/ui_lib/ui_nodes.py

```diff
@@ -2786,5 +3503,155 @@ def _init_rig_class_instance(self):
     CurveShapeItem.item_type: CurveShapeItem,
     ImportDataItem.item_type: ImportDataItem,
     PrintItem.item_type: PrintItem,
-    SetSkeletalMeshItem.item_type: SetSkeletalMeshItem
+    SetSkeletalMeshItem.item_type: SetSkeletalMeshItem,
+    GetSubControls.item_type: GetSubControls,
+    TransformVectorItem.item_type: TransformVectorItem
 }
+
+
+def update_socket_value(socket, update_rig=False, eval_targets=False):
```

Comment:
This needs to be broken apart. This function is way too long and needs to be broken apart into separate pieces.